### PR TITLE
[2.3-develop] Fixed issue #12514 Impossible to set source model when create new attribute

### PR DESCRIPTION
**Pull request by OleksiiStepanov**
### Description (*)

Pull request fix bug when unable create new attribute via api with custom source model
### Fixed Issues (if relevant)

1. magento/magento2#12514 : Impossible to set source model when create new attribute

### Manual testing scenarios (*)

create new attribute via api using \Magento\Catalog\Api\ProductAttributeRepositoryInterface
like this

```
$attribute = $this->attributeFactory->create();

$this->dataObjectHelper->populateWithArray(
$attribute,
[
'attribute_code' => 'select_attribute',
'source_model' => '\Custom\Model',
'default_frontend_label' => 'select_attribute_label',
'frontend_input' => 'select',
'global' => \Magento\Eav\Model\Entity\Attribute\ScopedAttributeInterface::SCOPE_STORE,
],
ProductAttributeInterface::class
);
$this->attributeRepository->save($attribute);
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19020) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @OleksiiStepanov. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by sivaschenko — APPROVED** [review]
Changes allow to set custom backend_type, source_model and backend_model when creating a new attribute. Tests could be styled and organized better, as in my opinion, they lack readability.

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-3376](https://jira.corp.magento.com/browse/ENGCOM-3376) has been created to process this Pull Request

File: dev/tests/integration/testsuite/Magento/Catalog/Model/Product/Attribute/RepositoryTest.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        );
+        $this->dataObjectHelper = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->get(
+            \Magento\Framework\Api\DataObjectHelper::class
+        );
+    }
+    public function testCustomSourceModel()
+    {
+        $attrCode = 'custom_source_model';
+        try {
+            $this->attributeRepository->deleteById($attrCode);
```
Comment (sidolov) [inline dev/tests/integration/testsuite/Magento/Catalog/Model/Product/Attribute/RepositoryTest.php:28]: Please, use data fixtures for creating and removing test data.

File: dev/tests/integration/testsuite/Magento/Catalog/Model/Product/Attribute/RepositoryTest.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
+<?php
```
Comment (sidolov) [inline dev/tests/integration/testsuite/Magento/Catalog/Model/Product/Attribute/RepositoryTest.php:1]: Please, fix code style in the whole file

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
Hi @OleksiiStepanov , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**